web/frontend/src/components/slam.vue

```diff
@@ -95,6 +105,84 @@ const fetchSLAMPose = (name: string): Promise<commonApi.Pose> => {
   });
 };
 
+const executeMoveOnMap = async () => {
+  moveClick = false;
+
+  /*
+   * set request name
+   * here we set the name of the motion service the user is using
+   */
+  motionServiceReq.setName('builtin');
+
+  const value = await fetchSLAMPose(props.name);
+  // set pose in frame
+  const destination = new commonApi.Pose();
+  destination.setX(Math.abs(destinationMarker.x - value.getX()));
+  destination.setY(Math.abs(destinationMarker.z - value.getY()));
+  destination.setZ(value.getZ());
+  destination.setOX(value.getOX());
+  destination.setOY(value.getOY());
+  destination.setOZ(value.getOZ());
+  destination.setTheta(value.getTheta());
+  motionServiceReq.setDestination(destination);
+
+  // set SLAM resource name
+  const slamResourceName = new commonApi.ResourceName();
+  slamResourceName.setNamespace('rdk');
+  slamResourceName.setType('service');
+  slamResourceName.setSubtype('slam');
+  slamResourceName.setName(props.name);
+  motionServiceReq.setSlamServiceName(slamResourceName);
+
+  // set component name
+  const baseResources = filterResources(props.resources, 'rdk', 'component', 'base');
+  if (baseResources === undefined) {
+    toast.error('No base component detected.');
+    return;
+  }
+  const baseResourceName = new commonApi.ResourceName();
+  baseResourceName.setNamespace('rdk');
+  baseResourceName.setType('component');
+  baseResourceName.setSubtype('base');
+  baseResourceName.setName(baseResources[0]!.name);
+  motionServiceReq.setComponentName(baseResourceName);
+
+  // set extra as position-only constraint
+  motionServiceReq.setExtra(
+    Struct.fromJavaScript({
+      motion_profile: "position_only"
+    })
+  )
+  
+  props.client.motionService.moveOnMap(
+    motionServiceReq,
+    new grpc.Metadata(),
+    (error: ServiceError | null, response: motionApi.MoveOnMapResponse | null) => {
+      if (error) {
+        toast.error(`Error moving: ${error}`);
+        return;
+      }
+      toast.success(`MoveOnMap success: ${response!.getSuccess()}`);
+    }
+  );
+
+};
+
+const executeStopMoveOnMap = () => {
+  moveClick = true;
+  props.client.motionService.moveOnMap(
+    motionServiceReq,
+    new grpc.Metadata(),
+    (error: ServiceError | null, response: motionApi.MoveOnMapResponse | null) => {
+      if (error) {
+        toast.error(`Error moving: ${error}`);
+        return;
+      }
+      toast.success(`Stopped MoveOnMap: ${response!.getSuccess()}`);
+    }
+  ).cancel
```

Comment:
What does the `cancel` do? This is just a property, not a function call right?

Reply:
`.cancel()` is my attempt at constructing a call which cancels the existng/executing motion.moveOnMap call.

I believe this **is** a function call since we send it out.

Reply:
Does this actually work? If so I don't understand how given that `.cancel` is just an attribute, not a function call & is not being assigned to any variable or returned from the function.